crates/services/txpool/src/service.rs

```diff
@@ -271,6 +273,53 @@ where
                 }
             }
 
+            // TODO: Refactor this to not block on calls to the peer and verification (need to open issue before merging)
+            new_peer_subscribed = self.new_tx_gossip_subscription.next() => {
```

Comment:
I think we need to do this change in this PR since this branch of Tokyo select can be very heavy and long. I think we need spawn a task for it.

We can create queue of task by using rayon for example. The same like we did with `crates/services/p2p/src/heavy_task_processor.rs` just to process this use case.

If you want, it can be another PR on top of this one, we will review and merge into this branch

Reply:
Why do we we want to use a rayon dedicated thread pool instead of spawning a call tokio task ?

Reply:
We can spawn tokyo task as well, we need to limit the number of active tasks to be one. Either we can write own queue of tasks and manage them, or reuse `rayon::spawn_fifo`.

Reply:
Implemented in : https://github.com/FuelLabs/fuel-core/pull/2131/commits/268d794bb0a70f0ccde2e0ad807f5a997c867538